apps/charterafrica/src/payload/fields/ecosystem/AirtableBaseSelect.js

```diff
@@ -0,0 +1,33 @@
+import { Select } from "payload/components/forms";
+import { select } from "payload/dist/fields/validations";
+import { createElement } from "react";
+import useSWR from "swr";
+
+import fetchJson from "../../../utils/fetchJson";
+
+const fetcher = (url) =>
+  fetchJson.get(url, { params: { source: "airtable", url: "/meta/bases" } });
+const url = `/api/v1/resources/ecosystem/proxy`;
+const getOptions = async () => {
+  const { bases } = await fetcher(url);
+  const options = bases.map((item) => ({ value: item.id, label: item.name }));
+  return options;
+};
+
+export const validateSelect = async (value, { hasMany, required, t }) => {
+  const options = await getOptions();
+  try {
+    return select(value, { hasMany, options, required, t });
+  } catch (error) {
+    return select(value, { hasMany, options, required, t });
+  }
+};
+
+function AirtableBaseSelect(props) {
```

Comment:
Not a component, should be named like other functions i.e. `airtableBaseSelect`

Reply:
it is a component @kilemensi only that it uses `React.createElement`.
Also uses React Hook `useSwr`

Reply:
Yo! I didn't see that @koechkevin. Why are we mixing naming-conventions then `AirtableBaseSelect` vs `airtableColumnSelect`?

Reply:
`airtableColumnSelect` is a function that builds a field but `AirtableBaseSelect` is only a field Component exported.
This informs the conventions

Reply:
Check the code sir... there is `columnSelect` which I assume is the component that `airtableColumnSelect` function uses, no?

Reply:
`columnSelect` That's a Higher Order Function that returns a component `ColumnSelect`

Reply:
Man this code gets complex by the minute! No way I'm going to remember how all of this fits together if somebody wakes me up in a middle of the night three months from now.

There has to be a simpler way... Let me think on it for a minute.